rs/state_machine_tests/src/lib.rs

```diff
@@ -3088,22 +3088,102 @@ impl StateMachine {
         Err(format!("No canister state for canister id {canister_id}."))
     }
 
-    /// Simulates a subnet split where the provided `canister_range` is assigned to a new subnet.
+    /// Produces a routing table, a canister migrations list and a subnet list record that reflects
+    /// splitting this subnet; then passes them to the registry data provider.
+    ///
+    /// Does not update the registry client on this subnet.
+    ///
+    /// This is intended to be used before calling `split` simulating a split of this subnet. Having
+    /// this as a separate step allows for other subnets to observe this update before this subnet
```

Comment:
How about after? While less likely, it can happen if the third party subnet is stalled / upgrading during the split.